strawberryfields/decompositions.py

```diff
@@ -450,22 +450,29 @@ def rectangular_symmetric(V, tol=1e-11):
         tuple[array]: returns a tuple of the form ``(tlist,np.diag(localV), None)``
             where:
 
-            * ``tlist``: list containing ``[n,m,internal_phase,external_phase,n_size]`` of the T unitaries needed
+            * ``tlist``: list containing ``[n, m, internal_phase, external_phase, n_size]`` of the T unitaries needed
             * ``localV``: Diagonal unitary matrix to be applied at the end of circuit
+            * ``None``: the value ``None``, in order to make the return
+              signature identical to the one of :func:`rectangular`.
     """
     tlist, diags, _ = rectangular_phase_end(V, tol)
     new_tlist, new_diags = [], np.ones(len(diags), dtype=diags.dtype)
     for i in tlist:
         em, en = int(i[0]), int(i[1])
         alpha, beta = np.angle(new_diags[em]), np.angle(new_diags[en])
         theta, phi = i[2], i[3]
-        external_phase = np.fmod((phi + alpha - beta), 2 * np.pi)
-        internal_phase = np.fmod((np.pi + 2.0 * theta), 2 * np.pi)
+        external_phase = (phi + alpha - beta) % (2 * np.pi)
+        internal_phase = (np.pi + 2.0 * theta) % (2 * np.pi)
+        # repeat modulo operations , otherwise the input unitary
+        # numpy.identity(20) yields an external_phase of exactly 2 * pi
```

Comment:
This problem seems to be well-known: see https://docs.python.org/3.7/reference/expressions.html#id17
    
> While abs(x%y) < abs(y) is true mathematically, for floats it may not be true numerically due to roundoff. For example, and assuming a platform on which a Python float is an IEEE 754 double-precision number, in order that -1e-100 % 1e100 have the same sign as 1e100, the computed result is -1e-100 + 1e100, which is numerically exactly equal to 1e100. The function math.fmod() returns a result whose sign matches the sign of the first argument instead, and so returns -1e-100 in this case. Which approach is more appropriate depends on the application.

Reply:
Good catch!